src/GitHub.TeamFoundation.14/Home/GitHubHomeSection.cs

```diff
@@ -65,7 +65,7 @@ public class GitHubHomeSection : TeamExplorerSectionBase, IGitHubHomeSection
                 timer.Tick += (s, e) =>
                 {
                     timer.Stop();
-                    if (!IsGitToolsMessageVisible())
+                    if (!IsGitToolsMessageVisible() && IsVisible)
```

Comment:
Instead of letting the timer tick and then check if it's visible here, move the whole if block to line 109 where we already have a `IsVisible` check and don't bother with the timer, that way we know for sure that we are really visible. This timer might kick in before VS triggers the event that makes us visible, so if we have this here, we might not show the message even if we're on a github repo, if VS happens to be slow.